### PR TITLE
lcd: remove unused structs

### PR DESCRIPTION
**Pull request by Lzw655**
# ESP-BSP Pull Request checklist

> Note: For new BSPs create a PR with this [link](https://github.com/espressif/esp-bsp/compare/main...my-branch?quick_pull=1&template=pr_template_bsp.md).

- [ ] Version of modified component bumped
- [ ] CI passing

# Change description
* ili9341 & st7796: remove unused structs
* st7796: fix the building error of test_apps when using esp32s2

### CODE REVIEW

File: components/lcd/esp_lcd_ili9341/idf_component.yml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-version: "2.0.0"
+version: "2.0.1"
```
Comment (tore-espressif) [inline components/lcd/esp_lcd_ili9341/idf_component.yml:1]: Let's not release this version. So no change in version here.

We don't want to release a new version after every change

File: components/lcd/esp_lcd_st7796/idf_component.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-version: "1.2.1"
+version: "1.2.2"
```
Comment (tore-espressif) [inline components/lcd/esp_lcd_st7796/idf_component.yml:1]: same here

File: components/lcd/esp_lcd_ili9341/idf_component.yml (hunk 1 above)
Comment (Lzw655) [reply components/lcd/esp_lcd_ili9341/idf_component.yml:1]: Okay, looks like there has been a change in our version modification principles.

**Comment by Lzw655** [comment]
@espzav PTAL, thank you!

**Review by espzav — APPROVED** [review]
LGTM

**Pull request merged**